hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,291 @@
+        part of its environment configuration.
+        """
+
+        # grab any base class settings
+        collector_settings = super(MariSessionCollector, self).settings or {}
+
+        # settings specific to this collector
+        mari_session_settings = {
+            "Work Template": {
+                "type": "template",
+                "default": None,
+                "description": "Template path for artist work files. Should "
+                               "correspond to a template defined in "
+                               "templates.yml. If configured, is made available"
+                               "to publish plugins via the collected item's "
+                               "properties. ",
+            },
+        }
+
+        # update the base settings with these settings
+        collector_settings.update(mari_session_settings)
+
+        return collector_settings
+
+    def process_current_session(self, settings, parent_item):
+        """
+        Analyzes the current session open in Mari and parents a subtree of
+        items under the parent_item passed in.
+
+        :param dict settings: Configured settings for this collector
+        :param parent_item: Root item instance
+        """
+
+        # create an item representing the current mari session
+        channel_item, layer_item = self.collect_current_mari_session(settings, parent_item)
+
+        self._collect_files(settings, channel_item, layer_item)
+
+    def collect_current_mari_session(self, settings, parent_item):
+        """
+        Creates an item that represents the current mari session.
+
+        :param dict settings: Configured settings for this collector
+        :param parent_item: Parent Item instance
+        :returns: Items of type mari.channels and mari.layers
+        """
+
+        if not mari.projects.current():
+            self.logger.warning("You must be in an open Mari project to be able to publish!")
+
+        publisher = self.parent
+
+        # create the channel item for the publish hierarchy
+        channel_item = parent_item.create_item(
+            "mari.channels",
+            "Publish flattened channels",
+            "Texture Channels",
+        )
+        # get the icon path to display for this item
+        icon_path = os.path.join(
+            self.disk_location,
+            os.pardir,
+            "icons",
+            "mari_channel_publish.png"
+        )
+        channel_item.set_icon_from_path(icon_path)
+
+        # create the layer item for the publish hierarchy
+        layer_item = parent_item.create_item(
+            "mari.layers",
+            "Publish individual layers for channels",
+            "Texture Channel Layers",
+        )
+        # get the icon path to display for this item
+        icon_path = os.path.join(
+            self.disk_location,
+            os.pardir,
+            "icons",
+            "mari_layer_publish.png"

```

Comment:
nitpick: For both of these items, we shouldn't really assume "publish". We should just collect the items (the nouns) and let the plugins attach themselves. They may publish the item or upload it or perform some other action. All that to suggest we shouldn't include an "action" in the filename.